cc.craftos_pc.CraftOS-PC.yaml

```diff
@@ -0,0 +1,120 @@
+    buildsystem: cmake-ninja
+    config-opts:
+      - "-DENABLE_ENCODINGS=OFF"
+      - "-DENABLE_ACTIVERECORD=OFF"
+      - "-DENABLE_ACTIVERECORD_COMPILER=OFF"
+      - "-DENABLE_MONGODB=OFF"
+      - "-DENABLE_DATA_SQLITE=OFF"
+      - "-DENABLE_REDIS=OFF"
+      - "-DENABLE_PROMETHEUS=OFF"
+      - "-DENABLE_PDF=OFF"
+      - "-DENABLE_ZIP=OFF"
+      - "-DENABLE_PAGECOMPILER=OFF"
+      - "-DENABLE_PAGECOMPILER_FILE2PAGE=OFF"
+    sources:
+      - type: archive
+        url: https://pocoproject.org/releases/poco-1.12.4/poco-1.12.4-all.tar.bz2
+        sha256: 4e6ccc0df904596478414ef56a992cf885eb4b0ee695b23d307eb2ff91c36626
+  
+  - name: libharu
+    buildsystem: cmake-ninja
+    sources:
+      - type: git
+        url: https://github.com/libharu/libharu.git
+        tag: v2.4.3
+        commit: 8dbcfe470581edc4bc52c3ef132d410e66cadc20
+  
+  - name: pngpp
+    buildsystem: simple
+    build-commands:
+      - make install PREFIX=/app
+    sources:
+      - type: archive
+        url: http://download.savannah.nongnu.org/releases/pngpp/png++-0.2.10.tar.gz
+        sha512: c54a74c0c20212bd0dcf86386c0c11dd824ad14952917ba0ff4c184b6547744458728a4f06018371acb7d5b842b641708914ccaa81bad8e72e173903f494ca85
+  
+  - shared-modules/SDL2/SDL2-with-libdecor.json
+  
+  # Needed at build-time only
+  - name: patchelf
+    buildsystem: autotools
+    cleanup:
+      - "*"
+    sources:
+      - type: git
+        url: https://github.com/NixOS/patchelf.git
+        tag: "0.17.2"
+        commit: 5908e16cd562bcb1909be4de0409c4912a8afc52
+  
+  - name: craftos2
+    buildsystem: simple
+    build-commands: 
+      - make -C craftos2-lua -j$FLATPAK_BUILDER_N_JOBS linux
+      - ./configure --without-ncurses --prefix=/app
+      - make -j$FLATPAK_BUILDER_N_JOBS
+
+      # https://github.com/MCJack123/craftos2/issues/237
+      - install -D -m 755 craftos2-lua/src/liblua.so /app/lib/libcraftos2-lua.so
+      - patchelf --replace-needed craftos2-lua/src/liblua.so libcraftos2-lua.so craftos
+
+      - install -D -m 755 craftos /app/bin/craftos
+    sources:
+      - type: git
+        url: https://github.com/MCJack123/craftos2.git
+        tag: v2.7.2
+        commit: e4ed6c0527d9983154df9b67d26593f84a4f2edc
+        x-checker-data:
+          type: git
+          tag-pattern: "^v([\\d.]+)$"
+
+  - name: craftos2-resources
+    buildsystem: simple
+    build-commands: 
+      - install -D -m 0644 CraftOS-PC.desktop "/app/share/applications/cc.craftos_pc.CraftOS-PC.desktop"
+      - desktop-file-edit --set-icon="cc.craftos_pc.CraftOS-PC" --set-key="Exec" --set-value="craftos" /app/share/applications/cc.craftos_pc.CraftOS-PC.desktop
+      - install -D -m 0644 256.png "/app/share/icons/hicolor/256x256/apps/cc.craftos_pc.CraftOS-PC.png"
+
+      - install -D -m 0644 appdata.xml /app/share/appdata/cc.craftos_pc.CraftOS-PC.appdata.xml
+      - sed -i 's/CraftOS-PC.desktop/cc.craftos_pc.CraftOS-PC.desktop/' /app/share/appdata/cc.craftos_pc.CraftOS-PC.appdata.xml
```

Comment:
if you install the desktop file with the original name and use `rename-desktop-file` you don't need to do that.